### PR TITLE
Topic/navbar refactory

### PR DESCRIPTION
**Pull request by pentateu**

### CODE REVIEW

File: src/models/NavigationBar.coffee [hunk 1]
```diff
@@ -119,31 +123,30 @@ class NavigationBar
...
               buttonParameters = buttonParametersFrom(button)
-              callback = button.onTap ? ->
+              params[location].push buttonParameters
 
               steroids.debug "steroids.navigationBar.setButtons adding button #{JSON.stringify(buttonParameters)} to location #{location}"
-              @buttonCallbacks[location].push callback
-              params[location].push buttonParameters
+
+              steroids.navigationBar.on "buttonTapped", (tapedButton) ->
+                if tapedButton.id == button.id
```
Comment (harsaharsa) [inline src/models/NavigationBar.coffee]: tapped button, not taped

File: src/models/NavigationBar.coffee [hunk 2]
```diff
@@ -119,31 +123,30 @@ class NavigationBar
...
-        for location in locations
+        for location in ["right", "left"]
           steroids.debug "steroids.navigationBar.setButtons constructing location #{location}"
-          @buttonCallbacks[location] = []
           params[location] = []
 
           if options[location]?
             for button in options[location]
               buttonParameters = buttonParametersFrom(button)
-              callback = button.onTap ? ->
```
Comment (harsaharsa) [inline src/models/NavigationBar.coffee]: can the onTap functionality be preserved on Steroids level too so existing code doesn't break? Or will we just keep compatibility on the Supersonic level and have the Steroids.js update be breaking?

**Comment by harsaharsa** [comment]
Should lines https://github.com/pentateu/steroids-js/blob/topic/navbar_refactory/src/models/NavigationBar.coffee#L152-L168 be removed to clean up code – they look like they're legacy android stuff?

File: src/models/NavigationBar.coffee (hunk 1 above)
Comment (pentateu) [inline src/models/NavigationBar.coffee]: thanks

File: src/models/NavigationBar.coffee [hunk 3]
```diff
@@ -228,34 +228,29 @@ class NavigationBar
...
-              callback = button.getCallback()
+              params.buttons[location].push parameters
 
               steroids.debug "steroids.navigationBar.update adding button #{JSON.stringify(parameters)} to location #{location}"
-              params.buttons[location].push parameters
-              @buttonCallbacks[location].push callback
+
+              steroids.navigationBar.on "buttonTapped", (tapedButton) ->
+                if tapedButton.id == button.id
+                  callback = button.getCallback() ? ->
```
Comment (pentateu) [inline src/models/NavigationBar.coffee]: here is where we preserve the button.onTap = () -> functionality.

**Comment by pentateu** [comment]
@harsaharsa 

I think the whole setButtons method should be made legacy and removed. .. but I'll check if we can remove the Android legacy stuff in the mean time.

**Pull request merged**